src/modm/driver/encoder/encoder_input.hpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021, Thomas Sommer
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_ENCODER_HPP
+#define MODM_ENCODER_HPP
+
+#include <modm/platform.hpp>
+#include <bit>
+#include <type_traits>
+
+namespace modm
+{
+/**
+ * \brief	Encoder
+ *
+ * \ingroup	modm_ui_encoder
+ * \author	2021 Thomas Sommer
+ */
+
+/**
+ * @ingroup modm_driver_encoder_input
+ * @author  Thomas Sommer
+ *
+ * @brief					This driver decodes a AB (incremental) encoder signal
+ *
+ * @tparam PH_A          	First modm::platform::Gpio pin to input the encoder signal.
+ * @tparam PH_B          	Second modm::platform::Gpio pin to input the encoder signal.
+ * @tparam DIVISION		 	Resolution: count of phasechange for one increment.
+ * @tparam UnderlyingType	Must be signed integer
+ */
+template<
+	typename PH_A,
+	typename PH_B,
+	uint8_t DIVISION = 4,
+	std::signed_integral UnderlyingType = int8_t
+	>
```

Comment:
For consistency: In bitbang_encoder_output the Gpio parameters are named <PinA, PinB, ...>

What's better or more modm'ish?

Reply:
I thought PH stood for Phase? Alternatively you can just call it SignalA and SignalB. We don't have a good naming convention for that…

Reply:
Yes it stands for Phase and i like it. bitbang_encoder_**output** its called PinA, PinB

I want to unify them.

Reply:
Ok i just call them all Signal.